server.js

```diff
@@ -19,26 +32,54 @@ app.get("/contacts", function(req, res)
        console.log("Connected to db.mean-tut-dev.contacts."); 
     });
     
-    var contactSchema = mongoose.Schema
+    var contacts = Contact.find({}).lean().exec(function(err, contact)
+    {
+        if(err)
+        {
+            console.log(err);    
+        }
+        
+        res.json(contact);
+    });
+    
+    mongoose.disconnect();
+});
+
+app.post("/contacts", function(req, res)
+{
+    mongoose.Promise = global.Promise;
+    mongoose.connect("mongodb://localhost/mean-tut-dev");
+    
+    var db = mongoose.connection;
+    db.on("error", console.error.bind(console, "connection error: "));
+    db.once("open", function()
+    {
+       console.log("Connected to db.mean-tut-dev.contacts."); 
+    });
+    
+    var contact = new Contact
     (
         {
-            name: String,
```

Comment:
Git is confusing this with the `Schema` block above. This is a separate block, in its own endpoint, which creates a new Contact object to save to the database.